test/unit/stringifies_method_signature_test.rb

```diff
@@ -0,0 +1,132 @@
+require "test_helper"
+
+module Mocktail
+  class StringifiesMethodSignatureTest < Minitest::Test
+    def setup
+      @subject = StringifiesMethodSignature.new
+    end
+
+    def test_basic_call
+      result = @subject.stringify(signature)
+
+      assert_equal "(&blk)", result
+    end
+
+    def test_positional_call
+      result = @subject.stringify(signature(
+        positional_params: Params.new(all: [:a, :b], required: [:a, :b])
+      ))
+
+      assert_equal "(a = ((__mocktail_default_args ||= {})[:a] = nil), b = ((__mocktail_default_args ||= {})[:b] = nil), &blk)", result
+    end
+
+    def test_optional_positional_call
+      result = @subject.stringify(signature(
+        positional_params: Params.new(
+          all: [:a, :b],
+          required: [:b],
+          optional: [:a]
+        )
+      ))
+
+      assert_equal "(a = ((__mocktail_default_args ||= {})[:a] = nil), b = ((__mocktail_default_args ||= {})[:b] = nil), &blk)", result
+    end
+
+    def test_kwarg_call
+      result = @subject.stringify(signature(
+        keyword_params: Params.new(
+          all: [:a, :b],
+          required: [:b],
+          optional: [:a]
+        )
+      ))
+
+      assert_equal "(a: ((__mocktail_default_args ||= {})[:a] = nil), b: ((__mocktail_default_args ||= {})[:b] = nil), &blk)", result
+    end
+
+    def test_block_call
+      result = @subject.stringify(signature(
+        block_param: :blocky
+      ))
+
+      assert_equal "(&blocky)", result
+    end
+
+    def test_argless_call
+      result = @subject.stringify(signature)
+
+      assert_equal "(&blk)", result
+    end
+
+    def test_rest_call
+      result = @subject.stringify(signature(
+        positional_params: Params.new(all: [:args], rest: :args)
+      ))
+
+      assert_equal "(*args, &blk)", result
+    end
+
+    def test_kwrest_call
+      result = @subject.stringify(signature(
+        keyword_params: Params.new(all: [:kwargs], rest: :kwargs)
+      ))
+
+      assert_equal "(**kwargs, &blk)", result
+    end
+
+    def test_complex_call
+      result = @subject.stringify(signature(
+        positional_params: Params.new(all: [:a, :b, :args], required: [:a], optional: [:b], rest: :args),
+        keyword_params: Params.new(all: [:c, :d, :kwargs], required: [:c], optional: [:d], rest: :kwargs),
+        block_param: :block
+      ))
+
+      assert_equal "(a = ((__mocktail_default_args ||= {})[:a] = nil), b = ((__mocktail_default_args ||= {})[:b] = nil), *args, c: ((__mocktail_default_args ||= {})[:c] = nil), d: ((__mocktail_default_args ||= {})[:d] = nil), **kwargs, &block)", result
+    end
+
+    unless Gem::Version.new(RUBY_VERSION) < Gem::Version.new("3.1")
+      eval(<<~RUBY, binding, __FILE__, __LINE__ + 1) # standard:disable Security/Eval
+        def dotdotdot(...)
+        end
+
+        def dotdotdot_with_args(a, b = :foo, ...)
+        end
+      RUBY
+    end
```

Comment:
```suggestion
    unless Gem::Version.new(RUBY_VERSION) < Gem::Version.new("3.1")
      def dotdotdot(...)
      end

      def dotdotdot_with_args(a, b = :foo, ...)
      end
    end
```

You can safely gate definitions behind conditionals, even if they contain (this) invalid syntax
```
 ❯ chruby 2.7
irb(main):001:1* if false
irb(main):002:1*   def dotdotdot(...); end
irb(main):003:0> end
=> nil
irb(main):004:0> 
Traceback (most recent call last):
        4: from /Users/caleb/.rubies/ruby-2.7.5/bin/irb:23:in `<main>'
        3: from /Users/caleb/.rubies/ruby-2.7.5/bin/irb:23:in `load'
        2: from /Users/caleb/.rubies/ruby-2.7.5/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        1: from (irb):7
NameError (undefined local variable or method `dotdotdot' for main:Object)
```